### ISSUE TITLE
UPDATE: end of Bard Macro Player ---> new app: Bard Music Player

### ISSUE THREAD

**Issue by pixaline** [parulina/bardmacroplayer#23]
Hello, apologies for the radio silence lately. Life has gotten busy so my focus has mostly been elsewhere other than this tool. However, it's very interesting to see how people have used it and the songs that have appeared for this.

However, the programming "language" (script) this tool is based on, AHK, is rather primitive and prone to errors. Truthfully, that wasn't a very thought out choice, as i instinctively went for it since I knew I was going to deal with key simulation. As it turns out, that is a very small part of the program, where most of the code deals with midi/file parsing and the player itself. This area is where AHK is really, really slow to develop in. So, i'm making the decision to archive this project. I won't be developing it further.

So, what'll happen to us bards?? Don't worry, because i'm making a sequel to replace it, called **Bard Music Player**. It's an entirely new tool written in C#, with many cool features such as memory reading, keyboard preview, orchestra mode, chat logging, etc. Making it a much better experience to parse, playback and manage songs. I've mostly finished making the program, about 90% done, all that remains is testing to iron out the bugs. Here's a screenshot:

![image](https://user-images.githubusercontent.com/6361957/43854155-581414da-9b42-11e8-9ad3-83e894b34a12.png)

When it's ready to be released, I will update this thread with links to download it.

### TL;DR: BardMacroPlayer is kill, I don't want to develop it anymore. It is replaced by BardMusicPlayer, a new application written in C# with advanced features. Please look forward to it

**Comment by RavenAmbree** [comment]
When will it be available for testing?
I have so many questions to ask about the program and it's capabilities, and I would also like to assist in documenting the features and how-to's.

**Comment by pixaline** [comment]
Right now i'm internally testing it among my friends. If you'd like to help me out with documentation/beta testing/feedback, feel free to join the [dev discord](https://discord.gg/mncXkGG) i've made. I also hang out at the [eorzea performing bards discord](https://www.reddit.com/r/ffxivperformances/comments/8yp552/join_eorzea_performing_bards_performance_network/), however i will only announce new program releases there.

**Comment by Decajack** [comment]
Awsome news, can't wait to try it!

**Comment by pixaline** [comment]
Update: the new program, Bard Music Player, has now been officially released at my website:
http://bmp.sqnya.se
The source for this program has now moved to a private bitbucket repo.
If you're interested in helping me develop, get in touch with me for repo access.

**Comment by HimitsuYami** [comment]
I can't seem to reach the page to download the new program. I get two different errors with two different browsers using the link you provided
![image](https://user-images.githubusercontent.com/21053554/47814792-f728af80-dd24-11e8-9635-11157cd3ef61.png)

**Comment by HimitsuYami** [comment]
I should add, that the version of Bard Macro Player I'm using (the latest one listed) works fine except for the fact that it's detecting one .mid file as 4 different tracks so when I want to play one I can't play the whole song

**Comment by Decajack** [comment]
@HimitsuYami That's because some songs have different instruments or playing 2 hands at the same time, so the macro player separates them like that.There's also some midis that play everything at the same time, but the song will sound clunky since the digital piano from FFXIV is limited. It depends how the midi song has been made tbh, hope it helps.
 EDIT: That's why parulina added the "Orchestra" option on the newest version, so the songs that have  4 or more tracks can be played with multiple players.
